server/src/test/java/org/kaazing/gateway/server/topic/StandaloneClusterTopicTest.java

```diff
@@ -143,53 +72,6 @@ public void shouldRejectAddAndRemoveFromMessageListenerSameThread() throws Inter
     }
 
     @Test
-    public void shouldAllowAddAndRemoveFromMessageListenerDifferentThread() throws InterruptedException {
-        ITopic<String> topic = factory.getTopic("topic_reject_add_remove_from_listener_different_thread");
-        CountDownLatch listenersCalled = new CountDownLatch(2);
-        MessageListener m1 = message -> listenersCalled.countDown();
-        String name = topic.addMessageListener(m1);
-        MessageListener m2 = message -> {
-            Thread t = new Thread(() -> {
-                topic.removeMessageListener(name);
-                listenersCalled.countDown();
-            });
-            t.start();
-            try {
-                t.join();
-            } catch (InterruptedException e) {
-                e.printStackTrace();
-            }
-        };
-        topic.addMessageListener(m2);
-        topic.publish("msg1");
-        listenersCalled.await();
-        assertEquals(1, topic.getLocalTopicStats().getPublishOperationCount());
-        assertEquals(2, topic.getLocalTopicStats().getReceiveOperationCount());
-        assertFalse(topic.removeMessageListener(name)); // already removed
-        topic.destroy();
-    }
-
-    @Test
-    public void shouldPubSubFromMessageListeners() throws InterruptedException {
-        CountDownLatch listenersCalled = new CountDownLatch(1);
-        ITopic<String> topic1 = factory.getTopic("topic1");
-        ITopic<String> topic2 = factory.getTopic("topic2");
-        MessageListener m1 = message -> topic2.publish("Message sent to topic2");
-        MessageListener m2 = message -> listenersCalled.countDown();
-        topic1.addMessageListener(m1);
-        topic2.addMessageListener(m2);
-        topic1.publish("trigger m2");
-        listenersCalled.await();
-        assertEquals(1, topic1.getLocalTopicStats().getPublishOperationCount());
-        assertEquals(1, topic1.getLocalTopicStats().getReceiveOperationCount());
-        assertEquals(1, topic2.getLocalTopicStats().getPublishOperationCount());
-        assertEquals(1, topic2.getLocalTopicStats().getReceiveOperationCount());
-        topic1.destroy();
-        topic2.destroy();
-    }
-
-
-    @Test
     public void shouldNotAllowNestedPublish() throws InterruptedException {
```

Comment:
You should catch the expected UnsupportedOperationException from the call to publish within the listener and set a variable to indicate the expected exception was thrown.  That would make the test more robust. As it is it could accidentally pass due to timing even if the exception was not thrown.